pkg/reconciler/taskrun/taskrun.go

```diff
@@ -603,17 +603,19 @@ func (c *Reconciler) createPod(tr *v1beta1.TaskRun, rtr *resources.ResolvedTaskR
 
 type DeletePod func(podName string, options *metav1.DeleteOptions) error
 
-func updateTaskRunResourceResult(taskRun *v1beta1.TaskRun, podStatus corev1.PodStatus) error {
+func updateTaskRunResourceResult(taskRun *v1beta1.TaskRun, pod corev1.Pod) error {
+	podconvert.SortContainerStatuses(&pod)
+
 	if taskRun.IsSuccessful() {
-		for idx, cs := range podStatus.ContainerStatuses {
+		for idx, cs := range pod.Status.ContainerStatuses {
 			if cs.State.Terminated != nil {
 				msg := cs.State.Terminated.Message
 				r, err := termination.ParseMessage(msg)
 				if err != nil {
 					return fmt.Errorf("parsing message for container status %d: %v", idx, err)
 				}
 				taskResults, pipelineResourceResults := getResults(r)
-				taskRun.Status.TaskRunResults = append(taskRun.Status.TaskRunResults, taskResults...)
+				taskRun.Status.TaskRunResults = replaceOrAppend(taskRun.Status.TaskRunResults, taskResults)
```

Comment:
I think we only need to do this once, after the loop, with the list of all results.  After line 621, something like:

```go
taskRun.Status.TaskRunResults = removeDuplicateResults(taskRun.Status.TaskRunResults)
```

(With `func replaceOrAppend` changed to `func removeDuplicateResults(results []v1beta1.TaskRunResult) []v1beta1.TaskRunResult`)

Does that make sense?  Once we've added all the duplicates to the list we can remove them in a single call.

Reply:
Thanks @sbwsg , I think it's make sense.